### DISCUSSION TITLE
Why every credentials members are transformed into string?

### DISCUSSION THREAD

**Discussion by EarthMessenger** [nextauthjs/next-auth#11473] [Help]
I'm using credentials as the login method. I pass boolean values and numbers into the `signIn` method with `signIn("credentials", { num: 1, str: "1", bool: true });`, but they become strings in the `authorize` method.

Is this the expected behavior? If so, is there an elegant way to verify that a string is a number using Zod?

You can see a reproduction of this issue here: https://github.com/EarthMessenger/authjs-credentials-report.git.

**Comment by Ekozmaster** [comment]
Hey there, I got the same issue today. Did you find a proper answer?
